### PR TITLE
🌍 Add PR template

### PR DESCRIPTION
**Pull request by jeffzh4ng**
Adds a basic PR template to the repo.

image

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Purpose
+_Describe the problem or feature in addition to a link to the issues._
+
+## Approach
```
Comment (xujustinj) [inline .github/pull_request_template.md:4]: on the more pragmatic side, we should add

## Testing
_Describe how the changes were tested._

**Review by xujustinj — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (jeffzh4ng) [reply .github/pull_request_template.md:4]: I like how you think sir — added.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+_How does this change address the problem?_
+
+## Testing
+_How did you test your changes?_
+
+#### Open Questions and Pre-Merge TODOs
+- [ ] Use github checklists. When solved, check the box and explain the answer.
+
+## Learning
```
Comment (xujustinj) [inline .github/pull_request_template.md:13]: I guess this is where we'd cite our StackOverflow and documentation sources

Comment (jeffzh4ng) [reply .github/pull_request_template.md:13]: Yep :-)

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+_Describe the research stage_
+
+_Links to blog posts, patterns, libraries or addons used to solve this problem_
+
+#### Blog Posts
```
Comment (xujustinj) [inline .github/pull_request_template.md:18]: Should we make this heading more general, or add similar sections for documentation / stackoverflow?

**Review by xujustinj — COMMENTED** [review]
lgtm otherwise

File: .github/pull_request_template.md (hunk 3 above)
Comment (jeffzh4ng) [reply .github/pull_request_template.md:18]: Hmmm, "Blog Posts" is meant to be a subheading for "Learning". I'll merge it in and we can update it if need be.

**Pull request merged**